privacyidea/lib/security/aeshsm.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+#
+#  2016-09-01 Mathias Brossard <mathias@axiadids.com>
+#             Alternate PKCS11 Security Module
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+import binascii
+import logging
+from privacyidea.lib.security.default import SecurityModule
+from privacyidea.lib.error import HSMException
+
+__doc__ = """
+This is a PKCS11 Security module that encrypts and decrypts the data on a HSM that is connected via PKCS11. This alternate version relies on AES keys.
+"""
+
+log = logging.getLogger(__name__)
+
+TOKEN_KEY = 0
+CONFIG_KEY = 1
+VALUE_KEY = 2
+
+mapping = {
+    'token':  TOKEN_KEY,
+    'config': CONFIG_KEY,
+    'value':  VALUE_KEY
+}
+
+try:
+    import PyKCS11
+except ImportError:
+    log.info("The python module PyKCS11 is not available. So we can not use the PKCS11 security module.")
+
+
+def int_list_to_bytestring(int_list):  # pragma: no cover
+    return "".join([chr(i) for i in int_list])
+
+
+class AESHardwareSecurityModule(SecurityModule):  # pragma: no cover
+
+    def __init__(self, config=None):
+        """
+        Initialize the PKCS11 Security Module.
+        The configuration needs to contain the pkcs11 module and the ID of the key.
+
+        {"module": "/usr/lib/hsm_pkcs11.so", "slot": 42, "key_label": "privacyidea"}
+
+        The HSM is not directly ready, since the HSM is protected by a password.
+        The function setup_module({"password": "HSM User password"}) needs to be called.
+
+        :param config: contains the HSM configuration
+        :type config: dict
+
+        :return: The Security Module object
+        """
+        config = config or {}
+        self.name = "HSM"
+        self.config = config
+        self.is_ready = False
+
+        if "module" not in config:
+            log.error("No PKCS11 module defined!")
+            raise HSMException("No PKCS11 module defined.")
+
+        label_prefix = config.get("key_label", "privacyidea")
+        self.key_labels = {}
+        for k in ['token', 'config', 'value']:
+            l = config.get(("key_label_%s" % k), None)
+            l = ('%s_%s' % (label_prefix, k)) if l is None else l
```

Comment:
The key_label_token, key_label_config, key_label_value, slot, password and module config options should be documented.

Can you please add the config file values to the docs with a new section "AES HSM" like the pkcs11 HSM?
https://github.com/privacyidea/privacyidea/blob/master/doc/installation/system/securitymodule.rst